DOCSP-28681: version update + retraction notice (#243)
mongodb/docs-golang: source/whats-new.txt

What's New in 1.11
--------------------

.. important:: Upgrade to Version 1.11.3 or Higher

>    Support for versions 1.11.0 through 1.11.2 has been retracted because

Reviewer: I: In Jira, Rachelle says we've retracted the versions themselves, not just support for the versions

S:

```suggestion
   Versions 1.11.0 through 1.11.2 of the driver have been retracted because
```
